### PR TITLE
Various build fixes from LEXCI in-person meeting

### PR DESCRIPTION
**Pull request by 20DM**
* Fixes typos and inconsistencies in README and Conan config file
* Make sure Conan passes on command-line options to CMake
* Disable MPI support by default
* Enables casacore by default
* Add custom logger
* Remove spdlog dependency

### CODE REVIEW

**Comment by 20DM** [comment]
Pipeline relies on https://github.com/astro-informatics/sopt/pull/422 going in first

File: cmake_files/dependencies.cmake
Comment (mmcleod89) [inline cmake_files/dependencies.cmake:1]: I'm a little confused as to how the FFTW openMP stuff is meant to work. If I turn on openMP when compiling then PURIFY_OPENMP_FFTW will be undefined in config.h and therefore it won't try to run FFTW with openMP. If it couldn't find openMP then we would run the line `set(PURIFY_OPENMP_FFTW FALSE)` which would mean that `PURIFY_OPENMP_FFTW` would be defined and set, so the `#ifdef PURIFY_OPENMP_FFTW` statements in the purify code would presumably evaluate to true even though OMP wasn't found?

Comment (20DM) [reply cmake_files/dependencies.cmake:1]: I didn't even realise until just now that this is what this is used for 😆  If we don't set it in the else part of the clause, it should work I guess.

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -134,17 +134,17 @@ jobs:
...
+        #  if [[ ${{matrix.cc}} == "clang" ]]; then
+        #    echo "compiler=apple-clang" >> ~/.conan2/profiles/default
+        #    echo "compiler.version=15" >> ~/.conan2/profiles/default
+        #    echo "compiler.libcxx=libstdc++11" >> ~/.conan2/profiles/default
+        #  else
+        #    echo "compiler=gcc" >> ~/.conan2/profiles/default
+        #    echo "compiler.version=11" >> ~/.conan2/profiles/default
+        #    echo "compiler.libcxx=libstdc++11" >> ~/.conan2/profiles/default
+        #  fi
+        #fi
```
Comment (mmcleod89) [inline .github/workflows/cmake.yml]: Are we keeping this as a comment because it might be useful later?

**Review by mmcleod89 — APPROVED** [review]

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (20DM) [reply .github/workflows/cmake.yml]: Ah, no, let's remove it. It was useful for debugging until I realised what the problem was 🤦

File: cmake_files/dependencies.cmake
Comment (20DM) [reply cmake_files/dependencies.cmake:1]: Thanks for catching the missing cstring include btw 👍

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (mmcleod89) [reply .github/workflows/cmake.yml]: Super, I'm happy for this to merge then if you are!

Comment (20DM) [reply .github/workflows/cmake.yml]: Let's go 🚀

**Pull request merged**